project-addons/crm_claim_rma_custom/crm_claim.py

```diff
@@ -61,6 +68,17 @@ def onchange_claim_type(self):
             return {'domain': {'partner_id': [('supplier', '=', True),
                                               ('is_company', '=', True)]}}
 
+    @api.multi
+    def write(self, vals):
+        if 'stage_id'in vals and vals['stage_id'] == 3:
+            for line in self.claim_line_ids:
+                line_state = line.substate_id.string_id
+                if line_state not in self.check_states:
```

Comment:
¿por qué no lo hacéis con los ids externos? ya se están utilizando en más sitios de la aplicación, abrís el modo desarrollador, entráis en los subestados y le dais a "ver metadatos" y ya sabéis cual es el id externo del subestado en el que estéis y no tenéis que crearos un campo nuevo ni rellenarlo a mano. Y además ya vais a tiro fijo, el if lo harías con un "==" en lugar de obligar a recorrerse una lista en cada iteracción.